pyopencl/array.py

```diff
@@ -66,7 +66,12 @@ def _get_common_dtype(obj1, obj2, queue):
 
     dsupport = has_double_support(queue.device)
     cache_key = None
-    o1_dtype = obj1.dtype
+    try:
+        o1_dtype = obj1.dtype
```

Comment:
How about 

`getattr(ob1, "dtype", type(obj1))`

Reply:
Doen in [3dd634e](https://github.com/inducer/pyopencl/pull/604/commits/3dd634e609a2cd69ca5e58c2177eb2b06f1ec5d1)

Reply:
Try and modify compyte to see whether that works out.

Reply:
> Try and modify compyte to see whether that works out.

With https://github.com/inducer/compyte/pull/40, this PR appears to pass the tests.